Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -0,0 +1,727 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <SFML/Network.hpp>
+
+#include "Common/BitUtils.h"
+#include "Common/Logging/Log.h"
+#include "Common/MsgHandler.h"
+#include "Core/HW/EXI/BBA/BuiltIn.h"
+#include "Core/HW/EXI/EXI_Device.h"
+#include "Core/HW/EXI/EXI_DeviceEthernet.h"
+
+namespace ExpansionInterface
+{
+u64 GetTickCountStd()
+{
+  using namespace std::chrono;
+  return duration_cast<milliseconds>(steady_clock::now().time_since_epoch()).count();
+}
+
+void SetHardwareInfo(u8* data, Common::MACAddress dest, Common::MACAddress src)
+{
+  Common::EthernetHeader* hwpart = (Common::EthernetHeader*)data;
+  *hwpart = Common::EthernetHeader(IP_PROTOCOL);
+  hwpart->destination = dest;
+  hwpart->source = src;
+}
+
+std::tuple<Common::EthernetHeader*, Common::IPv4Header*, Common::TCPHeader*>
+GetTCPHeaders(u8* data, Common::MACAddress dest, Common::MACAddress src)
+{
+  SetHardwareInfo(data, dest, src);
+  return std::tuple<Common::EthernetHeader*, Common::IPv4Header*, Common::TCPHeader*>(
+      (Common::EthernetHeader*)data, (Common::IPv4Header*)&data[14],
+      (Common::TCPHeader*)&data[0x22]);
+}
+
+std::tuple<Common::EthernetHeader*, Common::IPv4Header*, Common::UDPHeader*>
+GetUDPHeaders(u8* data, Common::MACAddress dest, Common::MACAddress src)
+{
+  SetHardwareInfo(data, dest, src);
+  return std::tuple<Common::EthernetHeader*, Common::IPv4Header*, Common::UDPHeader*>(
+      (Common::EthernetHeader*)data, (Common::IPv4Header*)&data[14],
+      (Common::UDPHeader*)&data[0x22]);
+}
```

Comment:
We will need to find an alternative to their uses as they don't take into accounts options.

Reply:
they are not used anymore, i havent cleaned the code yet